index.bs

```diff
@@ -6,16 +6,111 @@ Status: w3c/UD
 Group: Browser Testing and Tools Working Group
 URL: https://w3c.github.io/webdriver-bidi/
 Editor: Browser Testing and Tools Working Group, https://www.w3.org/testing/browser/
-No Editor: true
 Abstract: This document defines the BiDirectional WebDriver Protocol, a mechanism for remote control of user agents.
+Default Ref Status: current
 </pre>
 
 Introduction {#intro}
 =====================
 
-[[WEBDRIVER|WebDriver]] defines a protocol for introspection and remote control
-of user agents. This specification extends WebDriver by introducing
-bidirectional communication. In place of the strict command/response
-format of WebDriver, this permits events to stream from the user
-agent to the controlling software, better matching the evented nature
-of the browser DOM.
+[[WEBDRIVER2|WebDriver]] defines a protocol for introspection and
+remote control of user agents. This specification extends WebDriver by
+introducing bidirectional communication. In place of the strict
+command/response format of WebDriver, this permits events to stream
+from the user agent to the controlling software, better matching the
+evented nature of the browser DOM.
+
+Terminology {#terminology}
+==========================
+
+This specification reuses the terms defined in the
+[[!WEBDRIVER2]] specification.
+
+Transport {#transport}
+======================
+
+Message transport is provided using a [[RFC6455|WebSocket]]
+connection.
+
+A [[WEBDRIVER2#dfn-remote-ends|remote end]] has a
+<dfn>connection</dfn>, which is a network connection, and a
+<dfn>websocket url</dfn>, which is a string. These are both initially
+<code>null</code>.
+
+When data is recieved on [=connection=] it must be processed according
+to the requirements of the [[!RFC6455|WebSockets]]
+specification. When a [[RFC6455#section-6.2|WebSocket Message has
+been Received]], an implementation must [=handle an incoming
+message=]. When [[RFC6455#section-7.1.3|The WebSocket Closing
+Handshake is Started]] an implementation must [=end the
+session=].
```

Comment:
What happens if a socket prematurely dies? Do we need to care?

Reply:
I'm not sure we need to care; WebDriver 1 doesn't specify what happens if the server connection drops.

Reply:
I don't think we needed to care due to the command response nature and we didn't for `keep-alive`. I guess just wondering if it drops over a network(Think BrowserStack) do intermerdiary nodes need to restart and assume we're all good or  ...?

I don't think this needs to block right now but at least we need to be aware.

Reply:
At the moment I don't think we have any way to restart the connection if it drops. Making that actually work seems hard; you miss a load of events and it's hard to know how a client would actually cope. Can you file a separate issue on that?

Reply:
As an aside: I don't think using the bidi WebDriver protocol end-to-end over a network is going to be the best way to take advantage of bidi. Users could avoid a lot of potential reliability and latency issues by using bidi to communicate with a browser on a single machine, and building higher-level protocols on top of bidi. For example, rather than running a test script on a local machine and communicating back-and-forth with a remote end, a user might just submit their entire test script to the remote end to be executed on the same machine as the browser where the connection is much more reliable.

Reply:
@bwalderman that sounds like a good approach, but for large test suites already built on WebDriver that want to add some bidi-only functionality, running the bidi connection over a reliable network is likely what will happen. I've filed https://github.com/w3c/webdriver-bidi/issues/25 to keep track of this.